src/core/backend/pearl/group.rs

```diff
@@ -286,15 +290,13 @@ impl Group {
     pub fn create_pearl_holder(&self, start_timestamp: u64) -> Holder {
         let end_timestamp = start_timestamp + self.settings.timestamp_period_as_secs();
         let mut path = self.directory_path.clone();
-        path.push(format!("{}/", start_timestamp));
-
-        Holder::new(
-            start_timestamp,
-            end_timestamp,
-            self.vdisk_id,
-            path,
-            self.settings.config().clone(),
-        )
+        let hash = self.get_node_hash();
+        let partition_name = PartitionName::new(start_timestamp, &hash);
+        path.push(format!("{}/", partition_name.to_string()));
```

Comment:
If I remember correctly, you don't need `/` at path end. Anyway, it's better to push twice `partition_name` and `/`, instead of `format!()`.
(to avoid memory allocation)